sympy/concrete/products.py

```diff
@@ -279,11 +278,20 @@ def _eval_product(self, term, limits):
 
         elif term.is_Add:
             p, q = term.as_numer_denom()
-
-            p = self._eval_product(p, (k, a, n))
             q = self._eval_product(q, (k, a, n))
+            if q.is_Number:
 
-            return p / q
+                # There is expression, which couldn't change by
+                # as_numer_denom(). E.g. n**(2/3) + 1 --> (n**(2/3) + 1, 1).
+                # We have to catch this case.
+
+                p1, p2 = p.as_coeff_Add()
+                p1 = self._eval_product(p1, (k, a, n))
+                p2 = self._eval_product(p1, (k, a, n))
```

Comment:
Should the `p1` here be `p2`? If so, a test that would have failed with the current line could be added.